app/src/main/java/org/mozilla/tv/firefox/ext/Js.kt

```diff
@@ -249,4 +249,54 @@ var _firefoxTV_isPlaybackStateObserverLoaded;
     }
 })();
 """.trimIndent()
+
+    /**
+     * The soft keyboard sends `tab` events when `submit` is clicked. This adds an extra step
+     * to many user flows (for example, on google.com, people expect a search to kick off after
+     * clicking submit, but it doesn't). This script forces a `submit` call after tab events are
+     * received.
+     *
+     * This script will:
+     *  - Search the page for the largest (by element area) <input> element
+     *  - Look for its wrapping <form> element
+     *  - Attach a listener to the <input> to listen to key events while it is focused
+     *  - Call `submit` on the <form> whenever a `tab` key event is received
+     */
+    val ADD_SUBMIT_LISTENER_TO_MAIN_INPUT = """
```

Comment:
Note that this method changes substantially in the following commit.

Keeping this commit around in case we decide to use this more narrow behavior in the future.